src/ucai/core/utils/type_utils.py

```diff
@@ -86,3 +98,17 @@ def convert_timedelta_to_interval_str(time_val: datetime.timedelta) -> str:
     minutes, seconds = divmod(remainder, 60)
     microseconds = time_val.microseconds
     return f"INTERVAL '{days} {hours}:{minutes}:{seconds}.{microseconds}' DAY TO SECOND"
+
+
+def python_type_to_sql_type(py_type):
+    """Convert Python type to Unity Catalog SQL type."""
+    origin = get_origin(py_type)
+    if origin is dict:
+        return "MAP"
+    elif origin is list:
+        return "ARRAY"
```

Comment:
do we need to set the element type?

Reply:
yep and we need to recursively parse the container types. Updating with additional test coverage validation.